### ISSUE TITLE
Error not getting sent.

### ISSUE THREAD

**Issue by meetme2meat** [getsentry/sentry-go#394]
## Summary
I'm not entirely sure why this happens but I have seen this happening in past and it happening currently with one of our clients. Only after seeing the debug statement(i.e setting `Debug: true`) and looking sentry dashboard validate my suspicion that at times the events are not getting sent to the sentry server. 

Question:
Is `sentry.CaptureException` is async?

## Steps To Reproduce

Not sure if it will help but we have a very benign client that is supposed to report sentry error before the `log.Fatal` is written and the program exists because of it.

```go
   func main() {
	err := sentry.Init(sentry.ClientOptions{
		Debug: true,
		// ignore the nil error
		IgnoreErrors: []string{"Called CaptureException with nil value"},
	})
	if err != nil {
		log.Fatal(err)
	}

       err = someFunc()

	if err != nil {
		sentry.CaptureException(err) // this where the error occurs
		log.Fatal(err)
	}
  }
```
Following is the Debug output obtained for the sentry

```
[Sentry] 2021/12/14 15:03:37 Integration installed: ContextifyFrames
[Sentry] 2021/12/14 15:03:37 Integration installed: Environment
[Sentry] 2021/12/14 15:03:37 Integration installed: Modules
[Sentry] 2021/12/14 15:03:37 Integration installed: IgnoreErrors
[Sentry] 2021/12/14 15:03:37 Sending error event [2776ee1ea39e43de98525cd4b59425bb] to xxxx.ingest.sentry.io project: xxxxxx
{"level":"fatal","msg":"Replicas configuration can not be updated","time":"2021-12-14T15:03:37Z"}
```

No entry for following event id `2776ee1ea39e43de98525cd4b59425bb` on senty dashboard.

## Expected Behavior
The error sent should appear on the sentry dashboard. (mostly it does but as explained above are times it does not mention above is one such case)

### SDK
 - `sentry-go` version: v0.11.0 (as referred in go.mod)
 - Go version: We are building all binaries using `golang:alpine` image  
 - Using Go Modules? [yes/no]: yes

**Comment by rhcarvalho** [comment]
> Is sentry.CaptureException is async?

Depends on the transport being used. The default transport works in the background because that means calls to `sentry.CaptureError` are fast / do not depend on network speed.
But you can opt-in to a sync transport too.

Note that `log.Fatal` calls `os.Exit(1)` which terminates the program without any clean up / deferred calls, therefore Sentry events are not flushed.

https://pkg.go.dev/log#Fatal
https://pkg.go.dev/os#Exit

Solution:

You can either arrange your code such that only as few places call `os.Exit` / `log.Fatal` and then call `sentry.Flush` before exiting (example https://github.com/getsentry/sentry-go/blob/bd116d6ce79b604297c6497aa07d7ac01768adbb/example/basic/main.go#L48-L50), or opt-in to use the sync transport which will block and wait for each event to go through on every `sentry.CaptureError` call (example https://github.com/getsentry/sentry-go/blob/bd116d6ce79b604297c6497aa07d7ac01768adbb/example/synctransport/main.go#L17).
